src/core/lib/surface/call.cc

```diff
@@ -67,6 +67,10 @@
 
 #define MAX_SEND_EXTRA_METADATA_COUNT 3
 
+// These estimates are used to create arena for the first call.
+#define ESTIMATED_BATCH_CONTROL_COUNT 5
```

Comment:
Where does this number come from?  Do we have any data on the average number of batches per call?  If the goal here is to ensure that we always have enough space for all batches, then shouldn't we just use `MAX_CONCURRENT_BATCHES` instead?

More generally, I am concerned that we may be optimizing for the wrong case here.  While this change may benefit the specific use case of a service that has one streaming call per channel, unary RPCs are far more common than streaming, and unary RPCs typically have only 1-2 batches.  Will the hysteresis code ensure that we don't allocate more memory than we need after a few RPCs in the unary case?